types/types.go

```diff
@@ -107,12 +107,13 @@ type ImageDestination interface {
 	Reference() ImageReference
 	// FIXME? This should also receive a MIME type if known, to differentiate between schema versions.
 	PutManifest([]byte) error
-	// PutBlob writes contents of stream as a blob identified by digest.
+	// PutBlob writes contents of stream and returns its computed digest and size (both if can be computed).
```

Comment:
> and PutBlob(stream ReaderForDescriptor, digestIfKnown string) error.

that seems like an internal stuff in containers/image, I don't really want to expose that, OTOH, I'd like to return digests and sizes automagically (that's just a reader interface and as long as it is implemented we can compute those vars inside the library right?)